nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/wag/client/GenericApiGatewayClient.java

```diff
@@ -0,0 +1,134 @@
+        this.credentials = credentials;
+        this.apiKey = apiKey;
+        this.signer = new AWS4Signer();
+        this.signer.setServiceName(API_GATEWAY_SERVICE_NAME);
+        this.signer.setRegionName(region.getName());
+
+        final JsonOperationMetadata metadata = new JsonOperationMetadata().withHasStreamingSuccessResponse(false).withPayloadJson(false);
+        final Unmarshaller<GenericApiGatewayResponse, JsonUnmarshallerContext> responseUnmarshaller = in -> new GenericApiGatewayResponse(in.getHttpResponse());
+        this.responseHandler = SdkStructuredPlainJsonFactory.SDK_JSON_FACTORY.createResponseHandler(metadata, responseUnmarshaller);
+        JsonErrorUnmarshaller defaultErrorUnmarshaller = new JsonErrorUnmarshaller(GenericApiGatewayException.class, null) {
+            @Override
+            public AmazonServiceException unmarshall(JsonNode jsonContent) throws Exception {
+                return new GenericApiGatewayException(jsonContent.toString());
+            }
+        };
+        this.errorResponseHandler = SdkStructuredPlainJsonFactory.SDK_JSON_FACTORY.createErrorResponseHandler(
+                Collections.singletonList(defaultErrorUnmarshaller), null);
+
+        if (httpClient != null) {
+            super.client = httpClient;
+        }
+    }
+
+    public GenericApiGatewayResponse execute(GenericApiGatewayRequest request) {
+        return execute(request.getHttpMethod(), request.getResourcePath(), request.getHeaders(), request.getParameters(), request.getBody());
+    }
+
+    private GenericApiGatewayResponse execute(HttpMethodName method, String resourcePath, Map<String, String> headers, Map<String,List<String>> parameters, InputStream content) {
+        final ExecutionContext executionContext = buildExecutionContext();
+
+        DefaultRequest request = new DefaultRequest(API_GATEWAY_SERVICE_NAME);
+        request.setHttpMethod(method);
+        request.setContent(content);
+        request.setEndpoint(this.endpoint);
+        request.setResourcePath(resourcePath);
+        request.setHeaders(buildRequestHeaders(headers, apiKey));
+        if (parameters != null) {
+            request.setParameters(parameters);
+        }
+        return this.client.execute(request, responseHandler, errorResponseHandler, executionContext).getAwsResponse();
+    }
+
+    private ExecutionContext buildExecutionContext() {
+        final ExecutionContext executionContext = ExecutionContext.builder().withSignerProvider(
+                new DefaultSignerProvider(this, signer)).build();
+        executionContext.setCredentialsProvider(credentials);
+        executionContext.setSigner(signer);
+        return executionContext;
+    }
+
+    private Map<String, String> buildRequestHeaders(Map<String, String> headers, String apiKey) {
+        if (headers == null) {
+            headers = new HashMap<>();
+        }
+        if (apiKey != null) {
+            final Map<String, String> headersWithApiKey = new HashMap<>();
```

Comment:
Is it possible to just add the API key to `headers` without making a new map and then always `return headers;`?

Reply:
done